src/controller.js

```diff
@@ -48,18 +48,33 @@ function getKernelSource() {
   return `(${kernelSourceFunc})`;
 }
 
-// this feeds the SES realm's (real/safe) confineExpr() back into the Realm
+// this feeds the SES realm's (real/safe) makeCompartment() back into the Realm
 // when it does require('@agoric/evaluate'), so we can get the same
 // functionality both with and without SES
 function makeEvaluate(e) {
-  const { confineExpr, confine } = e;
-  const evaluateExpr = (source, endowments = {}, options = {}) =>
-    confineExpr(source, endowments, options);
-  const evaluateProgram = (source, endowments = {}, options = {}) =>
-    confine(source, endowments, options);
-  return Object.assign(evaluateExpr, {
-    evaluateExpr,
-    evaluateProgram,
+  const { makeCompartment, sesOptions } = e;
+  const { shims, transforms } = sesOptions;
+  const makeEvaluators = (realmOptions = {}) => {
+    const { shims: realmShims, transforms: realmTransforms } = realmOptions;
+    const c = makeCompartment({
```

Comment:
I can imagine the better flexibility this gets us.. are there any performance costs (to making a new compartment for each eval) that we should note?

Reply:
I changed this instead to create a new compartment only upon explicitly calling makeEvaluators (which is an application decision). The default evaluators are still confine/confineExpr.